pkg/controller/orchestrator/orchestrator_reconcile_test.go

```diff
@@ -0,0 +1,354 @@
+/*
+Copyright 2018 Pressinfra SRL
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package orchestratormysql
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	gomegatypes "github.com/onsi/gomega/types"
+
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+
+	api "github.com/presslabs/mysql-operator/pkg/apis/mysql/v1alpha1"
+	wrapcluster "github.com/presslabs/mysql-operator/pkg/controller/internal/mysqlcluster"
+	orc "github.com/presslabs/mysql-operator/pkg/orchestrator"
+	fakeOrc "github.com/presslabs/mysql-operator/pkg/orchestrator/fake"
+)
+
+var _ = Describe("MysqlCluster controller", func() {
+	var (
+		cluster   *wrapcluster.MysqlCluster
+		orcClient *fakeOrc.OrcFakeClient
+		rec       *record.FakeRecorder
+		orcSyncer Syncer
+	)
+
+	BeforeEach(func() {
+		rec = record.NewFakeRecorder(100)
+		orcClient = fakeOrc.New()
+		theCluster := &api.MysqlCluster{
+			ObjectMeta: metav1.ObjectMeta{Name: "the-name", Namespace: "the-namespace"},
+			Status: api.MysqlClusterStatus{
+				ReadyNodes: 1,
+			},
+			Spec: api.MysqlClusterSpec{
+				Replicas:   1,
+				SecretName: "a-name",
+			},
+		}
+		orcSyncer = NewOrcUpdater(theCluster, rec, orcClient)
+		cluster = wrapcluster.NewMysqlClusterWrapper(theCluster)
+	})
+
+	Describe("Update status from orc", func() {
```

Comment:
in test descriptions `orc` -> `orchestrator`.

Also we should make sure that test description flow is comfortable to read in plain English. For this, we should run test with
```sh
ginkgo --randomizeAllSpecs --randomizeSuites --failOnPending \
       --cover --coverprofile cover.out --trace --race -v \
       ./pkg/... ./cmd/...
```